### PR TITLE
Modify compute OKS function

### PR DESCRIPTION
**Pull request by shrivaths16**
### Description
Add method to calculate the OKS score based on the cocoeval API implementation in addition to implementation from the paper "Benchmarking and Error Diagnosis in Multi-Instance Pose Estimation." ICCV (2017).

### where `dx = xd - xg` (similarly for `dy`)

>---
>If keypoint visible
[`dx = xd - xg`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L222)
where
[`xg = g[0::3]; yg = g[1::3]; vg = g[2::3]  # The ground truth points`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L212)
[`xd = d[0::3]; yd = d[1::3]  # The detected points`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L219)
>
>---
>If keypoint not visible
[`dx = np.max((z, x0-xd),axis=0)+np.max((z, xd-x1),axis=0)  # `](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L227)
where
[`z = np.zeros((k))  # Ensures dx is never negative.`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L226)
[`x0 = bb[0] - bb[2]; x1 = bb[0] + bb[2] * 2  # x0 = 2 * xmin - xmax and x1 = 2 * xmax - xmin`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L215)
where bb
[`bb = gt['bbox']`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L214)
[`ann['bbox'] = [x0,y0,x1-x0,y1-y0]`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/coco.py#L359)
and [x0, x1](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/coco.py#L353-L356) come from the min/max keypoints
>```python
>                s = ann['keypoints']
>                x = s[0::3]
>                y = s[1::3]
>                x0,x1,y0,y1 = np.min(x), np.max(x), np.min(y), np.max(y)
>```

and
[`sigmas = p.kpt_oks_sigmas`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L205)
[`self.kpt_oks_sigmas = np.array([.26, .25, .25, .35, .35, .79, .79, .72, .72, .62,.62, 1.07, 1.07, .87, .87, .89, .89])/10.0`](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L523)

and [x0, x1](https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/coco.py#L353-L356) come from the min/max keypoints
>```python
>                s = ann['keypoints']
>                x = s[0::3]
>                y = s[1::3]
>                x0,x1,y0,y1 = np.min(x), np.max(x), np.min(y), np.max(y)
>```

### Types of changes

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
[list open issues here]

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [x] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: sleap/nn/evals.py [hunk 1]
```diff
@@ -145,6 +146,11 @@ def compute_oks(
             is the number of Euclidean dimensions (typically 2 or 3). Keypoints
             that are missing/not visible should be represented as NaNs.
         points_pr: Predicted instance of shape (n_pr, n_nodes, n_ed).
+        use_cocoeval: The use_cocoeval indicates whether the OKS score is calculated
```
Comment (roomrys) [inline sleap/nn/evals.py]:
```suggestion
        use_cocoeval: Indicates whether the OKS score is calculated 
```

File: sleap/nn/evals.py [hunk 2]
```diff
@@ -135,7 +135,8 @@ def compute_oks(
     points_gt: np.ndarray,
     points_pr: np.ndarray,
     scale: Optional[float] = None,
-    stddev: float = 0.025,
+    use_cocoeval: Optional[bool] = True,
```
Comment (roomrys) [inline sleap/nn/evals.py:138]:
```suggestion
    use_cocoeval: bool = True,
```
[`Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) indicates that `use_cocoeval` can also be `None`.

File: sleap/nn/evals.py [hunk 3]
```diff
@@ -203,11 +209,20 @@ def compute_oks(
...
+        normalization_factor = np.reshape(spread_factor, (1, 1, n_nodes)) * np.reshape(
+            scale_factor, (n_gt, 1, 1)
+        )
+    # If use_cocoeval is True, then compute normalization factor according to cocoeval.
+    else:
+        spread_factor = (2 * stddev) ** 2
+        scale_factor = 2 * (scale + np.spacing(1))
+        normalization_factor = np.reshape(spread_factor, (1, 1, n_nodes)) * np.reshape(
+            scale_factor, (n_gt, 1, 1)
+        )
```
Comment (roomrys) [inline sleap/nn/evals.py:225]:
```suggestion
    if use_cocoeval:
        # If use_cocoeval is True, then compute normalization factor according to cocoeval.
        spread_factor = (2 * stddev) ** 2
        scale_factor = 2 * (scale + np.spacing(1))
        normalization_factor = np.reshape(spread_factor, (1, 1, n_nodes)) * np.reshape(
            scale_factor, (n_gt, 1, 1)
        )
    else:
        # If use_cocoeval is False, then compute normalization factor according to the paper.
        spread_factor = stddev ** 2
        scale_factor = 2 * ((scale + np.spacing(1)) ** 2)
        normalization_factor = np.reshape(spread_factor, (1, 1, n_nodes)) * np.reshape(
            scale_factor, (n_gt, 1, 1)
        )
```

It should be pretty safe to use the bool stored in `use_cocoeval` for conditionals.

File: sleap/nn/evals.py [hunk 4]
```diff
@@ -145,6 +146,11 @@ def compute_oks(
             is the number of Euclidean dimensions (typically 2 or 3). Keypoints
             that are missing/not visible should be represented as NaNs.
         points_pr: Predicted instance of shape (n_pr, n_nodes, n_ed).
+        use_cocoeval: The use_cocoeval indicates whether the OKS score is calculated
```
Comment (roomrys) [inline sleap/nn/evals.py:149]:
```suggestion
        use_cocoeval: Indicates whether the OKS score is calculated
```

**Comment by talmo** [comment]
Can we add a permalink to the relevant lines in cocoutils for reference? Otherwise LGTM!

**Comment by shrivaths16** [comment]
> Can we add a permalink to the relevant lines in cocoutils for reference in the docstring? Otherwise LGTM!

I have added the permalink in the docstring of the compute_oks function.

**Pull request merged**